metadata-ingestion/scripts/avro_codegen.py

```diff
@@ -113,11 +112,85 @@ def make_load_schema_methods(schemas: Iterable[str]) -> str:
     )
 
 
+def annotate_aspects(aspects: List[dict], schema_class_file: Path) -> None:
+    schema_classes_lines = schema_class_file.read_text().splitlines()
+    line_lookup_table = {line: i for i, line in enumerate(schema_classes_lines)}
+
+    # Create the Aspect class.
+    # We ensure that it cannot be instantiated directly, as
+    # per https://stackoverflow.com/a/7989101/5004662.
+    schema_classes_lines[
+        # This is a no-op line, so it's safe to overwrite.
+        line_lookup_table["__SCHEMAS: Dict[str, RecordSchema] = {}"]
+    ] = """
+class _Aspect(DictWrapper):
+    ASPECT_NAME: str = None  # type: ignore
+
+    def __init__(self):
+        if type(self) is _Aspect:
+            raise TypeError("_Aspect is an abstract class, and cannot be instantiated directly.")
+        super().__init__()
+
+    @classmethod
+    def get_aspect_name(cls) -> str:
+        return cls.ASPECT_NAME  # type: ignore
+"""
+
+    for aspect in aspects:
+        className = f'{aspect["name"]}Class'
+        aspectName = aspect["Aspect"]["name"]
+        class_def_original = f"class {className}(DictWrapper):"
+
+        # Make the aspects inherit from the Aspect class.
+        class_def_line = line_lookup_table[class_def_original]
```

Comment:
If this class was already converted to use `_ASPECT` as base class, wouldn't this access throw a KeyError? Do we have a scenario where we incrementally update scema_classes.py?

Reply:
The codegen script always deletes and completely regenerates the schema_classes file, so I don't think this will be an issue